panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system_ros_interface.hpp

```diff
@@ -54,6 +59,36 @@ struct CANErrors
   bool rear_can_net_err;
 };
 
+class TriggerServiceWrapper
```

Comment:
We've attempted this previously, and it's quite challenging because the std::function parameter may or may not include the 'enable' parameter. It's difficult to pass callbacks with varying parameters and ensure that this parameter is correctly passed during callback invocation when using templates.